README.md
<p align="center"><img src="https://i.imgur.com/4mnFoTl.png" alt="White"></p>

Minimalist monochrome theme for [Visual Studio Code](https://code.visualstudio.com).

## Benefits

### Less UI, more code

It lets you focus on the beauty of your code by fading all visual distractions around.  
Perfect when you know your editor and its shortcuts.

### Meaningful syntax

All programming languages already have punctuation and style: we type characters, not rainbows.  
After some weeks away from noisy and inconsistent colorization, you'll start to have a better global view over your projects and switching between languages will be more natural.

### Knight & Day

After sunset, the Night version saves you from eye strain and improves the quality of your sleep.  
The [NightSwitch](https://marketplace.visualstudio.com/items?itemName=gharveymn.nightswitch) ([Lite](https://marketplace.visualstudio.com/items?itemName=gharveymn.nightswitch-lite)) extension can do the switch for you, every day on time.

## Screenshots

### White

<p align="center"><img src="https://i.imgur.com/DXMUjNT.png" alt="White screenshot"></p>

### White Night

<p align="center"><img src="https://i.imgur.com/asfhGzC.png" alt="White Night screenshot"></p>


CHANGELOG.md
# Changelog

## 1.3.6 - 2018-11-14

-   Add `keyword.operator.ternary`

## 1.3.5 - 2018-11-14

-   Add `storage.type.type`

## 1.3.4 - 2018-11-14

-   Add `support.type.object.module`

## 1.3.3 - 2018-11-14

-   Add `storage.type.tsx`

## 1.3.2 - 2018-11-13

-   Set new colors for Code 1.29 and debug

## 1.3.1 - 2018-11-09

-   Add Python directories to .vscodeignore

## 1.3.0 - 2018-11-05

-   Add Rust support

## 1.2.4 - 2018-06-14

-   Add `editorLineNumber.activeForeground`

## 1.2.3 - 2018-03-30

-   Add `editor.selectionHighlightBorder`

## 1.2.2 - 2018-03-29

-   Use real white for terminal white color

## 1.2.1 - 2018-03-19

-   Remove bold style of `meta.function.decorator`

## 1.2.0 - 2018-03-19

-   Generate short hexadecimal color values
-   Remove unused dependencies
-   Themes generated by Python instead of TypeScript

## 1.1.1 - 2018-03-12

-   Style peek view file and line foregrounds

## 1.1.0 - 2018-03-12

-   Style peek view

## 1.0.1 - 2018-03-08

-   Add scope `storage.type.namespace`

## 1.0.0 - 2018-03-06

-   Improved colors and languages support
-   Themes generated by code

## 0.0.4 - 2018-02-23

-   Improve integrated terminal colors

## 0.0.3 - 2018-02-23

-   Fix README typo

## 0.0.2 - 2018-02-23

-   Improved colors
-   More detailed README

## 0.0.1 - 2018-02-22

-   Complete UI customization
-   Good support for all trending languages

## 0.0.1-alpha.1 - 2018-02-21

-   Initial release


build.py
#!/usr/bin/env python3
"""Generate White and White Night themes."""

from __future__ import annotations
from copy import deepcopy
from typing import Any, Dict, Union, Type
import definitions
import json


class Color(object):
    """Color defines a color.

    Args:
        hex6 (str): The 6 digits hexadecimal representation.
        alpha (float, optional): The alpha value between 0 and 1. Defaults to 1.

    Raises:
        ValueError: The representation is not a 6 digits hexadecimal number.
    """

    def __init__(self, hex6: str, alpha: float = 1):
        if hex6.startswith("#"):
            hex6 = hex6[1:]
        if len(hex6) != 6:
            raise ValueError("Hexadecimal value must have 6 chars")
        hex6 = hex6.lower()

        self._rgb = int(hex6, 16)
        self.alpha = alpha

    @property
    def alpha(self) -> float:
        """Get the alpha of the color, always between 0 and 1.

        Returns:
            float: The alpha value.
        """
        return self._alpha

    @alpha.setter
    def alpha(self, v: float):
        """Get the alpha of the color.

        Args:
            v (float): The alpha value.

        Raises:
            ValueError: The value is not between 0 and 1.
        """
        if not 0 <= v <= 1:
            raise ValueError("Alpha must be between 0 and 1")
        self._alpha = v

    def __str__(self) -> str:
        """Get the shortest hexadecimal color value.

        Returns:
            str: The hexadecimal representation.
        """
        s = "{:06x}".format(self._rgb)
        if self.alpha < 1:
            s += "{:02x}".format(int(self.alpha * 255))
        if s[0] == s[1] and s[2] == s[3] and s[4] == s[5]:
            if len(s) == 6:
                s = s[0] + s[2] + s[4]
            elif s[6] == s[7]:
                s = s[0] + s[2] + s[4] + s[6]
        return "#" + s


class ColorSplit(object):
    """Contains colors for White and White Night themes.

    Args:
        white (Union[str, ColorSplit]): The White theme color from an hexadecimal
            representation or another color split.
        white_night (Union[str, ColorSplit], optional): The White Night theme color from
            an hexadecimal representation or another color split. If None, the White
            version is used. Defaults to None.
    """

    def __init__(
        self, white: Union[str, ColorSplit], white_night: Union[str, ColorSplit] = None
    ):
        if isinstance(white, ColorSplit):
            self.white: Color = deepcopy(white.white)
        else:
            self.white = Color(white)

        if white_night:
            if isinstance(white_night, ColorSplit):
                self.white_night: Color = deepcopy(white_night.white_night)
            else:
                self.white_night = Color(white_night)
        else:
            self.white_night = deepcopy(self.white)

    def set_alpha(self, white: float, white_night: float = None) -> ColorSplit:
        """Set alpha values of colors.

        Args:
            white (float): The alpha value for the White theme.
            white_night (float, optional): The alpha value for the White Night theme. If
                None, the White version is used. Defaults to None.

        Returns:
            ColorSplit: The same color split.
        """
        self.white.alpha = white
        self.white_night.alpha = white_night if white_night else white
        return self

    def with_alpha(self, white: float, white_night: float = None) -> ColorSplit:
        """Copy colors split and set alpha values of colors.

        Args:
            white (float): The alpha value for the White theme.
            white_night (float, optional): The alpha value for the White Night theme. If
                None, the White version is used. Defaults to None.

        Returns:
            ColorSplit: A color split copy.
        """
        color_split = deepcopy(self)
        color_split.set_alpha(white, white_night)
        return color_split


def forge(defs: Dict[str, Any], night: bool = False):
    """Replace color splits with plain strings.

    Args:
        defs (Dict[str, Any]): The definitions needed to be forged.
        night (bool, optional): Wether forge for the normal theme or its night version.
            Defaults to False.
    """
    for d in defs:
        if isinstance(defs[d], dict):
            forge(defs[d], night)
        elif isinstance(defs[d], list):
            for i, v in enumerate(defs[d]):
                if isinstance(v, dict):
                    forge(defs[d][i], night)
        elif isinstance(defs[d], definitions.ColorSplit):
            defs[d] = str(defs[d].white) if not night else str(defs[d].white_night)


def dump(night: bool = False):
    """Write definitions to JSON files.

    Args:
        night (bool, optional): Wether dump for the normal theme or its night version.
            Defaults to False.
    """
    filepath = "themes/White"
    if night:
        filepath += "-Night"
    filepath += "-color-theme.json"

    with open(filepath, "w") as f:
        defs = deepcopy(definitions.definitions)
        forge(defs, night)
        json.dump(defs, f, indent=4)


if __name__ == "__main__":
    dump()
    dump(True)


src/build.ts
import * as fs from "fs-extra";
import * as _ from "lodash";
import * as path from "path";
import { Color } from "./color";
import definitions from "./definitions";

function forgeColors(obj: any, night: boolean) {
    for (const prop in obj) {
        if (obj[prop] instanceof Color) {
            obj[prop] = night ? obj[prop].nightHex : obj[prop].hex;
        } else if (obj[prop] instanceof Object) {
            forgeColors(obj[prop], night);
        }
    }
    return obj;
}

function generate(night: boolean) {
    let filename = path.resolve(__dirname, "../themes/White");
    if (night) {
        filename += "-Night";
    }
    filename += "-color-theme.json";

    fs.outputJson(filename, forgeColors(_.cloneDeep(definitions), night), {
        spaces: 4,
    });
}

generate(false);
generate(true);


src/color.ts
import * as tinycolor from "tinycolor2";

export class Color {
    public color: string;
    public alpha: number;
    public nightColor: string;
    public nightAlpha: number;

    /**
     * Get the #rrggbbaa representation of the color.
     *
     * @returns string The color string.
     */
    get hex() {
        return tinycolor(this.color)
            .setAlpha(this.alpha)
            .toHex8String();
    }

    /**
     * Get the #rrggbbaa representation of the night variant of color.
     *
     * @returns string The color string.
     */
    get nightHex() {
        return tinycolor(this.nightColor)
            .setAlpha(this.nightAlpha)
            .toHex8String();
    }

    constructor(color: string | Color, nightColor?: string | Color) {
        if (color instanceof Color) {
            this.color = color.color;
            this.alpha = color.alpha;
            this.nightColor = color.nightColor;
            this.nightAlpha = color.nightAlpha;
        } else {
            this.color = this.nightColor = color;
            this.alpha = this.nightAlpha = 1;
        }

        if (nightColor instanceof Color) {
            this.nightColor = nightColor.nightColor;
            this.nightAlpha = nightColor.nightAlpha;
        } else if (typeof nightColor === "string") {
            this.nightColor = nightColor;
        }
    }

    /**
     * Set alpha for color.
     */
    public setAlpha(alpha: number, nightAlpha?: number): this {
        this.alpha = alpha;
        this.nightAlpha = nightAlpha !== undefined ? nightAlpha : alpha;
        return this;
    }

    /**
     * Get a copy of Color with alpha settings.
     *
     * @returns Color The new Color.
     */
    public withAlpha(alpha: number, nightAlpha?: number): Color {
        const copy = new Color(this.color, this.nightColor);
        copy.setAlpha(alpha, nightAlpha);
        return copy;
    }
}


src/definitions.ts
import { Color } from "./color";

// Colors

const none = new Color("#000000").setAlpha(0);
const white = new Color("#ffffff");
const black = new Color("#000000");
const grey = new Color("#bec9d3", "#2e2e37");
const greyStrong = new Color("#6b7a88", "#a8a8b1");
const greyLight = new Color("#e7ecf2", "#1e1d27");
const greyOnBlack = new Color("#65696e", grey);

const purple = new Color("#7d46fc");
const blue = new Color("#004bff");
const cyan = new Color("#00d2ff");
const green = new Color("#00ff68");
const yellow = new Color("#ffca00");
const orange = new Color("#ff9900");
const red = new Color("#ff0032");

const main = new Color(white, black);
const mainNegative = new Color(black, white);
const mainContrast = new Color("#f4f6fc", "#0a0b0f");
const mainContrastTransparent = new Color("#99a7e1", "#616a93").setAlpha(0.1);

const primary = new Color(blue, purple);
const primaryNegative = white;
const primaryHighlight = new Color("#0064ff", primary).setAlpha(0.15, 0.25);
const primaryHighlightStrong = primaryHighlight.withAlpha(0.3, 0.5);
const primaryHighlightLight = primaryHighlight.withAlpha(0.05, 0.08);

const added = green;
const addedHighlight = added.withAlpha(0.1, 0.2);
const deleted = red;
const deletedHighlight = deleted.withAlpha(0.1, 0.2);

const error = red;
const warning = orange;

const find = yellow.withAlpha(0.6);
const findHighlight = find.withAlpha(0.3);

const border = greyLight;

// Theme settings

export default {
    colors: {
        "activityBar.background": main,
        "activityBar.border": none,
        "activityBar.foreground": grey,
        "activityBarBadge.background": mainNegative,
        "activityBarBadge.foreground": main,
        "badge.background": mainNegative,
        "badge.foreground": main,
        "button.background": primary,
        "button.foreground": primaryNegative,
        "contrastBorder": none,
        "diffEditor.insertedTextBackground": addedHighlight,
        "diffEditor.removedTextBackground": deletedHighlight,
        "dropdown.background": mainContrast,
        "dropdown.border": border,
        "dropdown.foreground": mainNegative,
        "editor.background": main,
        "editor.findMatchBackground": find,
        "editor.findMatchHighlightBackground": findHighlight,
        "editor.foreground": mainNegative,
        "editor.lineHighlightBackground": primaryHighlightLight,
        "editor.lineHighlightBorder": none,
        "editor.selectionBackground": primaryHighlightStrong,
        "editor.selectionForeground": main,
        "editor.wordHighlightBackground": primaryHighlight,
        "editor.wordHighlightStrongBackground": primaryHighlightStrong,
        "editorBracketMatch.background": primaryHighlightStrong,
        "editorBracketMatch.border": none,
        "editorCursor.foreground": primary,
        "editorError.foreground": error,
        "editorGroup.border": mainContrast,
        "editorGroupHeader.tabsBackground": main,
        "editorGroupHeader.tabsBorder": none,
        "editorGutter.addedBackground": added,
        "editorGutter.deletedBackground": deleted,
        "editorGutter.modifiedBackground": mainNegative,
        "editorIndentGuide.background": mainContrast,
        "editorLineNumber.foreground": new Color("#ebf0f5", greyLight),
        "editorLink.activeForeground": primary,
        "editorOverviewRuler.addedForeground": none,
        "editorOverviewRuler.border": none,
        "editorOverviewRuler.bracketMatchForeground": primaryHighlightStrong,
        "editorOverviewRuler.errorForeground": error,
        "editorOverviewRuler.findMatchForeground": find,
        "editorOverviewRuler.modifiedForeground": mainNegative,
        "editorOverviewRuler.selectionHighlightForeground": primaryHighlightStrong,
        "editorOverviewRuler.warningForeground": warning,
        "editorOverviewRuler.wordHighlightForeground": primaryHighlight,
        "editorOverviewRuler.wordHighlightStrongForeground": primaryHighlightStrong,
        "editorRuler.foreground": mainContrast,
        "editorSuggestWidget.foreground": mainNegative,
        "editorWarning.foreground": warning,
        "editorWidget.background": mainContrast,
        "editorWidget.border": border,
        "errorForeground": error,
        "extensionButton.prominentBackground": primary,
        "extensionButton.prominentForeground": primaryNegative,
        "extensionButton.prominentHoverBackground": primary,
        "focusBorder": border,
        "foreground": mainNegative,
        "gitDecoration.ignoredResourceForeground": grey,
        "gitDecoration.modifiedResourceForeground": mainNegative,
        "gitDecoration.untrackedResourceForeground": mainNegative,
        "input.background": mainContrast,
        "input.border": border,
        "input.foreground": mainNegative,
        "input.placeholderForeground": grey,
        "list.activeSelectionBackground": mainNegative,
        "list.activeSelectionForeground": main,
        "list.dropBackground": primaryHighlightStrong,
        "list.focusBackground": primaryHighlight,
        "list.focusForeground": mainNegative,
        "list.highlightForeground": mainNegative,
        "list.hoverBackground": mainContrast,
        "list.inactiveSelectionBackground": mainContrast,
        "list.inactiveSelectionForeground": mainNegative,
        "panel.background": mainContrast,
        "panel.border": none,
        "panelTitle.activeBorder": grey,
        "peekView.border": mainNegative,
        "peekViewEditor.background": none,
        "peekViewEditor.matchHighlightBackground": find,
        "peekViewResult.background": none,
        "peekViewResult.fileForeground": greyStrong,
        "peekViewResult.lineForeground": greyStrong,
        "peekViewResult.matchHighlightBackground": find,
        "peekViewResult.selectionBackground": mainNegative,
        "peekViewResult.selectionForeground": main,
        "peekViewTitle.background": none,
        "peekViewTitleDescription.foreground": grey,
        "peekViewTitleLabel.foreground": mainNegative,
        "progressBar.background": primary,
        "scrollbar.shadow": none,
        "scrollbarSlider.background": mainContrastTransparent,
        "scrollbarSlider.hoverBackground": mainNegative,
        "sideBar.background": main,
        "sideBar.foreground": greyStrong,
        "sideBarSectionHeader.background": mainContrast,
        "sideBarSectionHeader.foreground": greyStrong,
        "sideBarTitle.foreground": grey,
        "statusBar.background": black,
        "statusBar.foreground": greyOnBlack,
        "statusBar.noFolderBackground": black,
        "tab.activeBackground": mainNegative,
        "tab.activeForeground": main,
        "tab.border": none,
        "tab.inactiveBackground": none,
        "tab.inactiveForeground": grey,
        "terminal.ansiBlack": mainNegative,
        "terminal.ansiBlue": blue,
        "terminal.ansiBrightBlack": mainNegative,
        "terminal.ansiBrightBlue": blue,
        "terminal.ansiBrightCyan": cyan,
        "terminal.ansiBrightGreen": green,
        "terminal.ansiBrightMagenta": purple,
        "terminal.ansiBrightRed": red,
        "terminal.ansiBrightWhite": greyStrong,
        "terminal.ansiBrightYellow": yellow,
        "terminal.ansiCyan": cyan,
        "terminal.ansiGreen": green,
        "terminal.ansiMagenta": purple,
        "terminal.ansiRed": red,
        "terminal.ansiWhite": greyStrong,
        "terminal.ansiYellow": yellow,
        "terminal.foreground": mainNegative,
        "terminal.selectionBackground": primaryHighlightStrong,
        "terminalCursor.foreground": primary,
        "textLink.activeForeground": primary,
        "textLink.foreground": primary,
        "titleBar.activeBackground": main,
        "titleBar.activeForeground": greyStrong,
        "titleBar.inactiveBackground": main,
        "titleBar.inactiveForeground": grey,
        "welcomePage.buttonBackground": mainContrast,
        "welcomePage.buttonHoverBackground": primaryHighlight,
        "widget.shadow": none,
    },
    tokenColors: [
        {
            scope: ["comment", "string.quoted.docstring"],
            settings: {
                foreground: grey,
            },
        },
        {
            scope: ["string"],
            settings: {
                foreground: greyStrong,
            },
        },
        {
            scope: [
                "punctuation.definition.string",
                "storage.type.string.python",
            ],
            settings: {
                foreground: mainNegative,
            },
        },
        {
            scope: [
                "beginning.punctuation",
                "entity.name.section.group-title",
                "entity.name.tag",
                "entity.other.attribute-name.class",
                "entity.other.attribute-name.id",
                "keyword.const",
                "keyword.control",
                "keyword.function",
                "keyword.import",
                "keyword.operator.assignment",
                "keyword.operator.comparison",
                "keyword.operator.decrement",
                "keyword.operator.expression",
                "keyword.operator.increment",
                "keyword.operator.increment-decrement",
                "keyword.operator.logical",
                "keyword.operator.new",
                "keyword.operator.other",
                "keyword.other.special-method",
                "keyword.package",
                "keyword.type",
                "keyword.var",
                "markup.heading",
                "meta.function.decorator",
                "meta.tag.sgml.doctype.html",
                "punctuation.separator.key-value",
                "storage.modifier",
                "storage.type.class",
                "storage.type.enum",
                "storage.type.function",
                "storage.type.import",
                "storage.type.interface",
                "storage.type.js",
                "storage.type.namespace",
                "storage.type.property",
                "storage.type.string.python",
                "storage.type.ts",
            ],
            settings: {
                fontStyle: "bold",
            },
        },
    ],
};


definitions.py
"""Themes color definitions."""

from build import ColorSplit

# Colors

colorNone = ColorSplit("#000000").set_alpha(0)
colorWhite = ColorSplit("#ffffff")
colorBlack = ColorSplit("#000000")
colorGrey = ColorSplit("#bec9d3", "#2e2e37")
colorGreyStrong = ColorSplit("#6b7a88", "#a8a8b1")
colorGreyLight = ColorSplit("#e7ecf2", "#1e1d27")
colorGreyOnBlack = ColorSplit("#65696e", colorGrey)

colorPurple = ColorSplit("#7d46fc")
colorBlue = ColorSplit("#004bff")
colorCyan = ColorSplit("#00d2ff")
colorGreen = ColorSplit("#00ff68")
colorYellow = ColorSplit("#ffca00")
colorOrange = ColorSplit("#ff9900")
colorRed = ColorSplit("#ff0032")

colorMain = ColorSplit(colorWhite, colorBlack)
colorMainNegative = ColorSplit(colorBlack, colorWhite)
colorMainContrast = ColorSplit("#f4f6fc", "#0a0b0f")
colorMainContrastTransparent = ColorSplit("#99a7e1", "#616a93").set_alpha(0.1)

colorPrimary = ColorSplit(colorBlue, colorPurple)
colorPrimaryNegative = colorWhite
colorPrimaryHighlight = ColorSplit("#0064ff", colorPrimary).set_alpha(0.15, 0.25)
colorPrimaryHighlightStrong = colorPrimaryHighlight.with_alpha(0.3, 0.5)
colorPrimaryHighlightLight = colorPrimaryHighlight.with_alpha(0.05, 0.08)

colorAdded = colorGreen
colorAddedHighlight = colorAdded.with_alpha(0.1, 0.2)
colorDeleted = colorRed
colorDeletedHighlight = colorDeleted.with_alpha(0.1, 0.2)

colorError = colorRed
colorWarning = colorOrange

colorFind = colorYellow.with_alpha(0.6)
colorFindHighlight = colorFind.with_alpha(0.3)

colorStackFocused = colorOrange.with_alpha(0.6)
colorStack = colorOrange.with_alpha(0.3)

colorBorder = colorGreyLight

# Theme definitions

definitions = {
    "colors": {
        "activityBar.background": colorMain,
        "activityBar.border": colorNone,
        "activityBar.foreground": colorGrey,
        "activityBarBadge.background": colorMainNegative,
        "activityBarBadge.foreground": colorMain,
        "badge.background": colorMainNegative,
        "badge.foreground": colorMain,
        "button.background": colorPrimary,
        "button.foreground": colorPrimaryNegative,
        "contrastBorder": colorNone,
        "debugToolBar.background": colorMainContrast,
        "diffEditor.insertedTextBackground": colorAddedHighlight,
        "diffEditor.removedTextBackground": colorDeletedHighlight,
        "dropdown.background": colorMainContrast,
        "dropdown.border": colorBorder,
        "dropdown.foreground": colorMainNegative,
        "editor.background": colorMain,
        "editor.findMatchBackground": colorFind,
        "editor.findMatchHighlightBackground": colorFindHighlight,
        "editor.focusedStackFrameHighlightBackground": colorStackFocused,
        "editor.foreground": colorMainNegative,
        "editor.lineHighlightBackground": colorPrimaryHighlightLight,
        "editor.lineHighlightBorder": colorNone,
        "editor.selectionBackground": colorPrimaryHighlightStrong,
        "editor.selectionForeground": colorMain,
        "editor.selectionHighlightBorder": ColorSplit(colorNone, colorPrimaryHighlight),
        "editor.stackFrameHighlightBackground": colorStack,
        "editor.wordHighlightBackground": colorPrimaryHighlight,
        "editor.wordHighlightStrongBackground": colorPrimaryHighlightStrong,
        "editorBracketMatch.background": colorPrimaryHighlightStrong,
        "editorBracketMatch.border": colorNone,
        "editorCursor.foreground": colorPrimary,
        "editorError.foreground": colorError,
        "editorGroup.border": colorMainContrast,
        "editorGroupHeader.tabsBackground": colorMain,
        "editorGroupHeader.tabsBorder": colorNone,
        "editorGutter.addedBackground": colorAdded,
        "editorGutter.deletedBackground": colorDeleted,
        "editorGutter.modifiedBackground": colorMainNegative,
        "editorIndentGuide.background": colorMainContrast,
        "editorLineNumber.activeForeground": colorGrey,
        "editorLineNumber.foreground": ColorSplit("#ebf0f5", colorGreyLight),
        "editorLink.activeForeground": colorPrimary,
        "editorOverviewRuler.addedForeground": colorNone,
        "editorOverviewRuler.border": colorNone,
        "editorOverviewRuler.bracketMatchForeground": colorPrimaryHighlightStrong,
        "editorOverviewRuler.errorForeground": colorError,
        "editorOverviewRuler.findMatchForeground": colorFind,
        "editorOverviewRuler.modifiedForeground": colorMainNegative,
        "editorOverviewRuler.selectionHighlightForeground": colorPrimaryHighlightStrong,
        "editorOverviewRuler.warningForeground": colorWarning,
        "editorOverviewRuler.wordHighlightForeground": colorPrimaryHighlight,
        "editorOverviewRuler.wordHighlightStrongForeground": colorPrimaryHighlightStrong,
        "editorRuler.foreground": colorMainContrast,
        "editorSuggestWidget.foreground": colorMainNegative,
        "editorWarning.foreground": colorWarning,
        "editorWidget.background": colorMainContrast,
        "editorWidget.border": colorBorder,
        "errorForeground": colorError,
        "extensionButton.prominentBackground": colorPrimary,
        "extensionButton.prominentForeground": colorPrimaryNegative,
        "extensionButton.prominentHoverBackground": colorPrimary,
        "focusBorder": colorBorder,
        "foreground": colorMainNegative,
        "gitDecoration.ignoredResourceForeground": colorGrey,
        "gitDecoration.modifiedResourceForeground": colorMainNegative,
        "gitDecoration.untrackedResourceForeground": colorMainNegative,
        "input.background": colorMainContrast,
        "input.border": colorBorder,
        "input.foreground": colorMainNegative,
        "input.placeholderForeground": colorGrey,
        "list.activeSelectionBackground": colorMainNegative,
        "list.activeSelectionForeground": colorMain,
        "list.dropBackground": colorPrimaryHighlightStrong,
        "list.focusBackground": colorPrimaryHighlight,
        "list.focusForeground": colorMainNegative,
        "list.highlightForeground": colorMainNegative,
        "list.hoverBackground": colorMainContrast,
        "list.inactiveSelectionBackground": colorMainContrast,
        "list.inactiveSelectionForeground": colorMainNegative,
        "panel.background": colorMainContrast,
        "panel.border": colorNone,
        "panelTitle.activeBorder": colorGrey,
        "peekView.border": colorMainNegative,
        "peekViewEditor.background": colorNone,
        "peekViewEditor.matchHighlightBackground": colorFind,
        "peekViewResult.background": colorNone,
        "peekViewResult.fileForeground": colorGreyStrong,
        "peekViewResult.lineForeground": colorGreyStrong,
        "peekViewResult.matchHighlightBackground": colorFind,
        "peekViewResult.selectionBackground": colorMainNegative,
        "peekViewResult.selectionForeground": colorMain,
        "peekViewTitle.background": colorNone,
        "peekViewTitleDescription.foreground": colorGrey,
        "peekViewTitleLabel.foreground": colorMainNegative,
        "progressBar.background": colorPrimary,
        "scrollbar.shadow": colorNone,
        "scrollbarSlider.background": colorMainContrastTransparent,
        "scrollbarSlider.hoverBackground": colorMainNegative,
        "sideBar.background": colorMain,
        "sideBar.foreground": colorGreyStrong,
        "sideBarSectionHeader.background": colorMainContrast,
        "sideBarSectionHeader.foreground": colorGreyStrong,
        "sideBarTitle.foreground": colorGrey,
        "statusBar.background": colorBlack,
        "statusBar.debuggingBackground": colorOrange,
        "statusBar.debuggingForeground": colorBlack,
        "statusBar.foreground": colorGreyOnBlack,
        "statusBar.noFolderBackground": colorBlack,
        "tab.activeBackground": colorMainNegative,
        "tab.activeForeground": colorMain,
        "tab.activeModifiedBorder": colorNone,
        "tab.border": colorNone,
        "tab.inactiveBackground": colorNone,
        "tab.inactiveForeground": colorGrey,
        "tab.inactiveModifiedBorder": colorMainNegative,
        "terminal.ansiBlack": colorMainNegative,
        "terminal.ansiBlue": colorBlue,
        "terminal.ansiBrightBlack": colorMainNegative,
        "terminal.ansiBrightBlue": colorBlue,
        "terminal.ansiBrightCyan": colorCyan,
        "terminal.ansiBrightGreen": colorGreen,
        "terminal.ansiBrightMagenta": colorPurple,
        "terminal.ansiBrightRed": colorRed,
        "terminal.ansiBrightWhite": colorWhite,
        "terminal.ansiBrightYellow": colorYellow,
        "terminal.ansiCyan": colorCyan,
        "terminal.ansiGreen": colorGreen,
        "terminal.ansiMagenta": colorPurple,
        "terminal.ansiRed": colorRed,
        "terminal.ansiWhite": colorWhite,
        "terminal.ansiYellow": colorYellow,
        "terminal.foreground": colorMainNegative,
        "terminal.selectionBackground": colorPrimaryHighlightStrong,
        "terminalCursor.foreground": colorPrimary,
        "textLink.activeForeground": colorPrimary,
        "textLink.foreground": colorPrimary,
        "titleBar.activeBackground": colorMain,
        "titleBar.activeForeground": colorGreyStrong,
        "titleBar.inactiveBackground": colorMain,
        "titleBar.inactiveForeground": colorGrey,
        "welcomePage.buttonBackground": colorMainContrast,
        "welcomePage.buttonHoverBackground": colorPrimaryHighlight,
        "widget.shadow": colorNone,
    },
    "tokenColors": [
        {
            "scope": ["comment", "string.quoted.docstring"],
            "settings": {"foreground": colorGrey},
        },
        {"scope": ["string"], "settings": {"foreground": colorGreyStrong}},
        {
            "scope": ["punctuation.definition.string", "storage.type.string.python"],
            "settings": {"foreground": colorMainNegative},
        },
        {
            "scope": [
                "beginning.punctuation",
                "entity.name.section.group-title",
                "entity.name.tag",
                "entity.other.attribute-name.class",
                "entity.other.attribute-name.id",
                "keyword.const",
                "keyword.control",
                "keyword.function",
                "keyword.import",
                "keyword.operator.assignment",
                "keyword.operator.comparison",
                "keyword.operator.decrement",
                "keyword.operator.expression",
                "keyword.operator.increment",
                "keyword.operator.increment-decrement",
                "keyword.operator.logical",
                "keyword.operator.misc",
                "keyword.operator.new",
                "keyword.operator.other",
                "keyword.operator.ternary",
                "keyword.other.fn",
                "keyword.other.rust",
                "keyword.other.special-method",
                "keyword.other.where",
                "keyword.package",
                "keyword.type",
                "keyword.var",
                "markup.heading",
                "meta.tag.sgml.doctype.html",
                "punctuation.separator.key-value",
                "storage.modifier",
                "storage.type.class",
                "storage.type.enum",
                "storage.type.function",
                "storage.type.import",
                "storage.type.interface",
                "storage.type.js",
                "storage.type.namespace",
                "storage.type.property",
                "storage.type.rust",
                "storage.type.string.python",
                "storage.type.ts",
                "storage.type.tsx",
                "storage.type.type",
                "support.type.object.module",
            ],
            "settings": {"fontStyle": "bold"},
        },
    ],
}
